File: projectintegrationFront/src/app/service/Representation/representation.service.ts
import { AddRepresentation, UpdateRepresentation} from './../../class/representation';
import { Injectable } from '@angular/core';
import { Representation } from '../../class/representation';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Command, addCommand } from '../../class/command';

@Injectable({
  providedIn: 'root'
})
export class RepresentationService {

 ApiUri: string ='https://localhost:44364/api/Representation';
  constructor(public http: HttpClient) { }

  get(): Observable<Representation[]>
  {
    return this.http.get<Representation[]>(this.ApiUri+'/api/Representation');
  }
   getbyid(id:number): Observable<Representation>
  {
    return this.http.get<Representation>(`${this.ApiUri}/${id}`);
  }
  create(data: AddRepresentation):void
  {

    this.http.post<AddRepresentation>(this.ApiUri, data);
  }
  update(data: UpdateRepresentation):void
  {
     this.http.put<UpdateRepresentation>(this.ApiUri, data);
  }
  delete(id:number):void
  {
    this.http.delete<Representation>(`${this.ApiUri}/${id}`);
  }
  //---update par rapport au modification
  getSalle(idsalle: number): Observable<Representation[]> {
    return this.http.get<Representation[]>(`${this.ApiUri}/get-salle/${idsalle}`);
  }
   getPiece(idPiece: number): Observable<Representation[]> {
     return this.http.get<Representation[]>(`${this.ApiUri}/get-piece/${idPiece}`);
  }
  addCommandRepresentation(id:number,data: addCommand): void
  {
    this.http.post<addCommand>(`${this.ApiUri}/add-command/${id}`,data);
  }
    deleteCommandRepresentation(idrep:number,idCommand:number):void
  {
    this.http.delete<Representation>(`${this.ApiUri}/delete-command/${idrep}/${idCommand}`);
  }
}

File: ProjecIntegration.Api/Blazor.UI/data/services/RepresentationService.cs
using Blazor.UI.modelViews;
using System.Net.Http.Json;

namespace Blazor.UI.services
{
    public interface IRepresentationService
    {

        Task<IEnumerable<RepresentationDto>> Get();
        Task<RepresentationDto> GetById(int id);
        Task Create(AddRepresentationDto data);
        Task Update(UpdateRepresentationDto data);
        Task Delete(int id);
        Task<IEnumerable<RepresentationDto>> GetSalle(int idSalle);
        Task<IEnumerable<RepresentationDto>> GetPiece(int idPiece);
        Task AddCommandRepresentation(int id, AddCommandDto data);
        Task DeleteCommandRepresentation(int idRep, int idCommand);
    }
   
        public class RepresentationService : IRepresentationService
        {
            private readonly HttpClient _httpClient;
            private const string ApiUri = "https://localhost:44337/api/Representation";

            public RepresentationService(HttpClient httpClient)
            {
                _httpClient = httpClient;
            }

            public async Task<IEnumerable<RepresentationDto>> Get()
            {
                return await _httpClient.GetFromJsonAsync<IEnumerable<RepresentationDto>>(ApiUri);
            }

            public async Task<RepresentationDto> GetById(int id)
            {
                return await _httpClient.GetFromJsonAsync<RepresentationDto>($"{ApiUri}/{id}");
            }

            public async Task Create(AddRepresentationDto data)
            {
                await _httpClient.PostAsJsonAsync(ApiUri, data);
            }

            public async Task Update(UpdateRepresentationDto data)
            {
                await _httpClient.PutAsJsonAsync(ApiUri, data);
            }

            public async Task Delete(int id)
            {
                await _httpClient.DeleteAsync($"{ApiUri}/{id}");
            }

            public async Task<IEnumerable<RepresentationDto>> GetSalle(int idSalle)
            {
                return await _httpClient.GetFromJsonAsync<IEnumerable<RepresentationDto>>($"{ApiUri}/get-salle/{idSalle}");
            }

            public async Task<IEnumerable<RepresentationDto>> GetPiece(int idPiece)
            {
                return await _httpClient.GetFromJsonAsync<IEnumerable<RepresentationDto>>($"{ApiUri}/get-piece/{idPiece}");
            }

            public async Task AddCommandRepresentation(int id, AddCommandDto data)
            {
                await _httpClient.PostAsJsonAsync($"https://localhost:44337/api/Representation/add-command/{id}", data);
            }

            public async Task DeleteCommandRepresentation(int idRep, int idCommand)
            {
                await _httpClient.DeleteAsync($"{ApiUri}/delete-command/{idRep}/{idCommand}");
            }
        } 
 }
